pandas/core/internals/api.py

```diff
@@ -114,6 +123,16 @@ def make_block(
         dtype = dtype or values.dtype
         klass = get_block_type(dtype)
 
+    elif klass is _DatetimeTZBlock and not isinstance(values.dtype, DatetimeTZDtype):
+        # pyarrow calls get here (pyarrow<15)
```

Comment:
This goes together with temporarily adding back DatetimeTZBlock as mentioned above in https://github.com/pandas-dev/pandas/pull/61723/files#r2179667396, and updated the comment to indicate this is only needed as long as we support pyarrow < 15